src/common/ChannelChatters.cpp

```diff
@@ -64,9 +64,31 @@ void ChannelChatters::addPartedUser(const QString &user)
         });
     }
 }
+
+const QColor ChannelChatters::getUserColor(const QString &user)
```

Comment:
Unless I'm missing something here, I think we should prevent unnecessary copies and return a `const QColor &` instead. The function can be `const` as well

Reply:
I don't know the threading model that is used in chatterino, but since the color map is behind a mutex, I'll leave it as is.

Reply:
Thanks for getting back to the suggestions so quickly.

Would you mind explaining what your concern in this particular case is?
(Not saying there is none by the way :smile: Interested in learning something though)

Reply:
My concern is that if the color map gets updated after the reference is returned and before it is displayed, the reference could point to the wrong data.

Compiler also warned against returning a reference - since I don't do a lot of C++ I don't know how references work, but I'm assuming it's just like pointers.

Reply:
This section in the [`std::map::insert_or_assign`](https://en.cppreference.com/w/cpp/container/map/insert_or_assign) leads me to think otherwise:

> No iterators or references are invalidated.

Also, I did not get any compiler warnings with gcc when I tried the change, are you using clang?

(Also, the function could be `const` in either case)